### PR TITLE
RAR: Log full path of bad-image failures

### PR DESCRIPTION
**Pull request by rainersigwald**
When RAR is trying to find the closure of dependencies of the input
assemblies, it searches for assemblies based on identity in paths next
to currently known assemblies. Today, if there is a file of the expected
name that is not a well-formed managed assembly, RAR will silently
ignore it, without reporting any details about the file.

This changes the RAR output format

```
Dependency "DotLiquid, Version=2.0.174.0, Culture=neutral, PublicKeyToken=82e46016ecf9f07c".
    Could not find dependent files. Could not load file or assembly 'DotLiquid.dll' or one of its dependencies. An attempt was made to load a program with an incorrect format.
    Could not resolve this reference. Could not locate the assembly "DotLiquid, Version = 2.0.174.0, Culture=neutral, PublicKeyToken=82e46016ecf9f07c". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
    For SearchPath "S:\repro\".
        Considered "S:\repro\\DotLiquid.winmd", but it didn't exist.
    Required by "Microsoft.Online.DirectoryServices.Management.dll".
```

To

```
Dependency "DotLiquid, Version=2.0.174.0, Culture=neutral, PublicKeyToken=82e46016ecf9f07c".
    Could not resolve this reference. Could not locate the assembly "DotLiquid, Version = 2.0.174.0, Culture=neutral, PublicKeyToken=82e46016ecf9f07c". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
    For SearchPath "S:\repro\".
        Considered "S:\repro\\DotLiquid.winmd", but it didn't exist.
        Considered "S:\repro\\DotLiquid.dll", which existed but didn't have a valid identity. This may not be an assembly.
        Considered "S:\repro\\DotLiquid.exe", but it didn't exist.
    For SearchPath "{TargetFrameworkDirectory}".
        Considered "C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\DotLiquid.winmd", but it didn't exist.
        Considered "C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\DotLiquid.dll", but it didn't exist.
        Considered "C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\DotLiquid.exe", but it didn't exist.
        Considered "C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\DotLiquid.winmd", but it didn't exist.
        Considered "C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\DotLiquid.dll", but it didn't exist.
        Considered "C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\DotLiquid.exe", but it didn't exist.
    Required by "Microsoft.Online.DirectoryServices.Management.dll".
```

### CODE REVIEW

**Comment by danmoseley** [comment]
As a user I'm not sure what "didn't have a valid identity" means. Eg., it sounds like strong naming. Would it be better to simply say "does not appear to be a valid .NET assembly" or something?

**Comment by rainersigwald** [comment]
> Would it be better to simply say "does not appear to be a valid .NET assembly" or something?

Yeah, I don't think that loses any relevant info.

**Comment by rainersigwald** [comment]
I've backed this down to just the improved error handling and the improved string (and removed all my horrible workarounds/experiments).

File: src/Tasks/AssemblyDependency/Resolver.cs [hunk 1]
```diff
@@ -214,6 +214,14 @@ ResolutionSearchLocation searchLocation
 
                     // ...falling through and relying on the targetAssemblyName==null behavior below...
                 }
+                catch (BadImageFormatException)
+                {
+                    // As above, this is weird: there's a valid reference to an assembly with a file on disk
+                    // that isn't a valid .NET assembly. Might be the result of mid-build corruption, but
```
Comment (Forgind) [inline src/Tasks/AssemblyDependency/Resolver.cs:220]: valid .NET assembly == managed not native? Are there other requirements?

**Review by Forgind — COMMENTED** [review]
Is this expected to be non-fatal? I'm wondering if it's really better to not surface an exception as far down the stack as possible.

**Review by Forgind — APPROVED** [review]

**Pull request merged**